server/asset/eth/rpcclient.go

```diff
@@ -110,11 +187,12 @@ func (c *rpcclient) connect(ctx context.Context) (err error) {
 
 	var success bool
 
-	c.clients = make([]*ethConn, len(c.endpoints))
-	for i, endpoint := range c.endpoints {
+	c.clients = make([]*ethConn, 0, len(c.endpoints))
+	for _, endpoint := range c.endpoints {
 		client, err := rpc.DialContext(ctx, endpoint)
 		if err != nil {
-			return fmt.Errorf("unable to dial rpc to %q: %v", endpoint, err)
+			c.log.Errorf("Ethereum RPC client failed to connect to %q: %v", endpoint, err)
+			continue
 		}
```

Comment:
I get this is the point of the pr, but I think it should still keep this failed node around and check periodically. Say an owner has a geth node that just isn't synced yet, and for whatever reason just doesn't respond here, or the server starts up with the node offline, they might not notice it was not added if they had a fallback and it did connect. You wouldn't want to keep your server down waiting for the eth node to act right. I don't think they can add again after starting? Which may be a use for the admin api endpoint at some point. But for now I think it should keep checking those.

The health checking loop does keep them around.

Reply:
Actually, would market suspend -> resume attempt to connect all the nodes again?

Reply:
It appears not to.

Reply:
> I get this is the point of the pr, but I think it should still keep this failed node around and check periodically.

Yeah I agree, it wouldn't hurt to attempt to connect once every few minutes.